pad.c

```diff
@@ -2809,6 +2978,27 @@ C<PADNAMEf_OUTER> flag already set.
 =cut
 */
 
+{
+    struct padname_with_str *alloc;
+    char *alloc2; /* for Newxz */
+    PADNAME *pn;
+    STRLEN len = name_len + sizeof (symbol_table);
+    PERL_ARGS_ASSERT_NEW_PADNAME_SYMBOL_PVN;
+
+    Newxz(alloc2,
+          STRUCT_OFFSET(struct padname_with_str, xpadn_str[0]) + len + 1,
+          char);
+    alloc = (struct padname_with_str *)alloc2;
```

Comment:
Why Newxz() when 4 lines below Copy() will write over the null bytes? No need for interp backend or libc to do a redundant `memset()` .

Reply:
it is current implementation modified with treating symbol table and name as separated arguments.